Assets/MixedRealityToolkit/Inspectors/MixedRealityWireframeShaderGUI.cs

```diff
@@ -0,0 +1,210 @@
+            customRenderingMode = FindProperty(BaseStyles.customRenderingModeName, props);
+            sourceBlend = FindProperty(BaseStyles.sourceBlendName, props);
+            destinationBlend = FindProperty(BaseStyles.destinationBlendName, props);
+            blendOperation = FindProperty(BaseStyles.blendOperationName, props);
+            depthTest = FindProperty(BaseStyles.depthTestName, props);
+            depthWrite = FindProperty(BaseStyles.depthWriteName, props);
+            depthOffsetFactor = FindProperty(BaseStyles.depthOffsetFactorName, props);
+            depthOffsetUnits = FindProperty(BaseStyles.depthOffsetUnitsName, props);
+            colorWriteMask = FindProperty(BaseStyles.colorWriteMaskName, props);
+
+            cullMode = FindProperty("_CullMode", props);
+            renderQueueOverride = FindProperty("_RenderQueueOverride", props);
+
+            baseColor = FindProperty("_BaseColor", props);
+            wireColor = FindProperty("_WireColor", props);
+            wireThickness = FindProperty("_WireThickness", props);
+        }
+
+        public override void AssignNewShaderToMaterial(Material material, Shader oldShader, Shader newShader)
+        {
+            float? cullMode = GetFloatProperty(material, "_Cull");
+
+            base.AssignNewShaderToMaterial(material, oldShader, newShader);
+   
+            SetFloatProperty(material, null, "_CullMode", cullMode);
+
+            // Setup the rendering mode based on the old shader.
+            if (oldShader == null || !oldShader.name.Contains("Legacy Shaders/"))
+            {
+                SetupMaterialWithRenderingMode(material, (RenderingMode)material.GetFloat(BaseStyles.renderingModeName), CustomRenderingMode.Opaque, -1);
+            }
+            else
+            {
+                RenderingMode mode = RenderingMode.Opaque;
+
+                if (oldShader.name.Contains("/Transparent/Cutout/"))
+                {
+                    mode = RenderingMode.TransparentCutout;
+                }
+                else if (oldShader.name.Contains("/Transparent/"))
+                {
+                    mode = RenderingMode.Transparent;
+                }
+
+                material.SetFloat(BaseStyles.renderingModeName, (float)mode);
+
+                MaterialChanged(material);
+            }
+        }
+
+        protected void Initialize(Material material)
+        {
+            if (!initialized)
+            {
+                MaterialChanged(material);
+                initialized = true;
+            }
+        }
+
+        protected void MaterialChanged(Material material)
+        {
+            SetupMaterialWithRenderingMode(material, 
+                (RenderingMode)renderingMode.floatValue, 
+                (CustomRenderingMode)customRenderingMode.floatValue, 
+                (int)renderQueueOverride.floatValue);
+        }
+
+        // TODO: Merge this out? with StandardShaderGUI
```

Comment:
> // TODO: Merge this out? with StandardShaderGUI [](start = 7, length = 48)

Nit: Avoid having dangling TODOs that don't have more background/value/stuff.

If you don't see any value in merging this and don't think it will ever matter just get rid of the TODO. If there is value it might be worth just doing now.

Reply:
yes, since I am just going to move the materialproperties per my other comment, I'll just move this out as well. Both classes should be able to share this.